### PR TITLE
INT-1408: Implement SSH tunnel connector

### PR DESCRIPTION
**Pull request by durran**
This change is [Reviewable](https://reviewable.io/reviews/mongodb-js/data-service/16)

### CODE REVIEW

File: lib/ssh-tunnel-connector.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  /**
+   * Connect to the SSH tunnel and execute the callback.
+   *
+   * @param {Function} callback - The callback.
+   */
+  connect(callback) {
+    if (this.options.host) {
+      debug('Found SSH tunnel options, opening SSH tunnel.');
+      return tunnel(this.options, function(result, error) {
```
Comment (rueckstiess) [inline lib/ssh-tunnel-connector.js]: Would we get incremental status from this object? Like the connection steps from the design drafts?

File: lib/data-service.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
 'use strict';
 
 const debug = require('debug')('mongodb-data-service:data-service');
+const Connection = require('mongodb-connection-model');
```
Comment (rueckstiess) [inline lib/data-service.js]: Where is `Connection` used in the new code below?

File: lib/ssh-tunnel-connector.js (hunk 1 above)
Comment (durran) [inline lib/ssh-tunnel-connector.js]: No unfortunately the tunnel function does not give us that - I am looking at options to bake that in.

File: lib/data-service.js (hunk 2 above)
Comment (durran) [inline lib/data-service.js]: I need to add a check if the model passed is not a `Connection` to instantiate a new one.

**Pull request merged**